pkg/platform/model/buildplanner/impactreport.go

```diff
@@ -0,0 +1,50 @@
+package buildplanner
+
+import (
+	"encoding/json"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/pkg/buildscript"
+	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/request"
+	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/response"
+)
+
+type buildScripter interface {
+	BuildScript() *buildscript.BuildScript
+}
+
+type ImpactReportParams struct {
+	Owner   string
+	Project string
+	Before  buildScripter
```

Comment:
Using `*buildplanner.Commit` would result in an import cycle.

Reply:
Assuming there isn't an import cycle with buildscripts you could pass in the buildscript instead? It looks like that's all you're accessing on the commit.

I'd like to find a way to address this without interfaces if we can find a feasible solution.